### DISCUSSION TITLE
write holding register

### DISCUSSION THREAD

**Discussion by drakesoft** [StephanJoubert/home_assistant_solarman#365] [Q&A]
Hello,
is it possible to write holding register with home_assistant_solarman? I want to shutdown/Start my deye hybrid inverter with writing to addr 80.

**Comment by StephanJoubert** [comment]
Hi @drakesoft ,  the latest unreleased code includes the service call to do that. It is currently in testing and once I get the green light will draft a new release (hopefully by the end of the week)

 If you feel up to it, you can check it out and copy manually to Home Assistant to test.

**Reply by UndIch** [reply]
Sorry - obviously I am to stupid to find it... did you publish this code somewhere or did you sent it to @drakesoft privately? there is no such thing like a developer branch here right?

**Reply by drakesoft** [reply]
Just download the main branch.

**Comment by drakesoft** [comment]
Hi @StephanJoubert, 
I tried but  it don't worked as estimated.
1.  I get the error "No socket available".  [log.txt](https://github.com/StephanJoubert/home_assistant_solarman/files/12306917/log.txt)
2.  I have 4  devices in config how can I choose the inverter or ip?
Maybe it tries to connect to a wrong inverter.  Some are offline right now.
By the way with a short pysolarmanv5 script it works.

**Comment by drakesoft** [comment]
With the unreleased code my 3 Deye SUN1600 is not working anymore. I switched back to release version.
[home-assistant_2023-08-10T09-39-53.613Z.log](https://github.com/StephanJoubert/home_assistant_solarman/files/12311400/home-assistant_2023-08-10T09-39-53.613Z.log)

**Comment by UndIch** [comment]
So tried it out - doesn’t seem to work for my Deye800G3. Doesn’t find any of the entities. To bad, would have loved the option to actually change something.
